code/datums/preferences.dm

```diff
@@ -918,6 +918,16 @@ $(function() {
 			if (jobban_isbanned(user,J.name) || (J.needs_college && !user.has_medal("Unlike the director, I went to college")) || (J.requires_whitelist && !NT.Find(ckey(user.mind.key))))
 				src.jobs_unwanted += J.name
 				continue
+			if (J.rounds_needed_to_play)
+				if (user.client)
+					var/ckey = user.client.ckey
+					var/list/response = null
+					try
+						response = apiHandler.queryAPI("playerInfo/get", list("ckey" = ckey), forceResponse = 1)
+					catch
+					if (response && (response["participated"]) < J.rounds_needed_to_play) //if theyve played less rounds than the job needs, unwanted it
```

Comment:
I'm reasonably sure the response does indeed return a number but just in case it returns a string for some reason this is the place to debug later.